stage1_fly/run/main.go

```diff
@@ -0,0 +1,381 @@
+	discardBool    bool
+	discardString  string
+)
+
+func getHostMounts() (map[string]struct{}, error) {
+	hostMounts := map[string]struct{}{}
+
+	mi, err := os.Open("/proc/self/mountinfo")
+	if err != nil {
+		return nil, err
+	}
+	defer mi.Close()
+
+	sc := bufio.NewScanner(mi)
+	for sc.Scan() {
+		var (
+			discard    string
+			mountPoint string
+		)
+
+		_, err := fmt.Sscanf(sc.Text(),
+			"%s %s %s %s %s",
+			&discard, &discard, &discard, &discard, &mountPoint,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		hostMounts[mountPoint] = struct{}{}
+	}
+	if sc.Err() != nil {
+		return nil, fmt.Errorf("problem parsing mountinfo: %v", sc.Err())
+	}
+	return hostMounts, nil
+}
+
+func init() {
+	flag.BoolVar(&debug, "debug", false, "Run in debug mode")
+
+	// The following flags need to be supported by stage1 according to
+	// https://github.com/coreos/rkt/blob/master/Documentation/devel/stage1-implementors-guide.md
+	// TODO: either implement functionality or give not implemented warnings
+	flag.Var(&discardNetlist, "net", "Setup networking")
+	flag.BoolVar(&discardBool, "interactive", true, "The pod is interactive")
+	flag.StringVar(&discardString, "mds-token", "", "MDS auth token")
+	flag.StringVar(&discardString, "local-config", common.DefaultLocalConfigDir, "Local config path")
+}
+
+func lookupPath(bin string, paths string) (string, error) {
+	pathsArr := filepath.SplitList(paths)
+	for _, path := range pathsArr {
+		binPath := filepath.Join(path, bin)
+		binAbsPath, err := filepath.Abs(binPath)
+		if err != nil {
+			return "", fmt.Errorf("unable to find absolute path for %s", binPath)
+		}
+		d, err := os.Stat(binAbsPath)
+		if err != nil {
+			continue
+		}
+		// Check the executable bit, inspired by os.exec.LookPath()
+		if m := d.Mode(); !m.IsDir() && m&0111 != 0 {
+			return binAbsPath, nil
+		}
+	}
+	return "", fmt.Errorf("unable to find %q in %q", bin, paths)
+}
+
+func withClearedCloExec(lfd int, f func() error) error {
+	err := sys.CloseOnExec(lfd, false)
+	if err != nil {
+		return err
+	}
+	defer sys.CloseOnExec(lfd, true)
+
+	return f()
+}
+
+func writePpid(pid int) error {
+	// write ppid file as specified in
+	// Documentation/devel/stage1-implementors-guide.md
+	out, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("Cannot get current working directory: %v\n", err)
+	}
+	// we are the parent of the process that is PID 1 in the container so we write our PID to "ppid"
+	err = ioutil.WriteFile(filepath.Join(out, "ppid"),
+		[]byte(fmt.Sprintf("%d\n", pid)), 0644)
+	if err != nil {
+		return fmt.Errorf("Cannot write ppid file: %v\n", err)
+	}
+	return nil
+}
+
+func evaluateMounts(rfs string, app string, p *stage1commontypes.Pod) ([]flyMount, error) {
+	imApp := p.Images[app].App
+	namedVolumeMounts := map[types.ACName]volumeMountTuple{}
+
+	for _, m := range p.Manifest.Apps[0].Mounts {
+		_, exists := namedVolumeMounts[m.Volume]
+		if exists {
+			log.Fatalf("fly: duplicated mount given: %q", m.Volume)
```

Comment:
Why `log.Fatalf` instead of just returning an error? Also, if returning an error, remove the "fly: " prefix.